app/assets/javascripts/discourse/app/models/topic-tracking-state.js

```diff
@@ -182,17 +182,11 @@ const TopicTrackingState = EmberObject.extend({
   },
 
   dismissNewTopic(data) {
-    Object.keys(this.states).forEach((k) => {
-      const topic = this.states[k];
-      if (
-        (!data.payload.category_id ||
-          topic.category_id === parseInt(data.payload.category_id, 10)) &&
-        (!data.payload.tag_id || topic.tags.includes(data.payload.tag_id))
-      ) {
-        this.states[k] = Object.assign({}, topic, {
-          is_seen: true,
-        });
-      }
+    data.payload.topic_ids.forEach((k) => {
+      const topic = this.states[`t${k}`];
+      this.states[`t${k}`] = Object.assign({}, topic, {
+        is_seen: true,
+      });
```

Comment:
this logic is now simplified as message receive exact topic_ids which should be marked as "seen"